provider/cluster_rosa_classic_resource.go

```diff
@@ -1178,10 +1203,72 @@ func (r *ClusterRosaClassicResource) Update(ctx context.Context, request tfsdk.U
 		)
 		return
 	}
-	diags = response.State.Set(ctx, state)
+	diags = response.State.Set(ctx, plan)
 	response.Diagnostics.Append(diags...)
 }
 
+func updateProxy(state, plan *ClusterRosaClassicState, clusterBuilder *cmv1.ClusterBuilder) (*cmv1.ClusterBuilder, bool, error) {
+	shouldUpdateProxy := false
+	if (state.Proxy == nil && plan.Proxy != nil) || (state.Proxy != nil && plan.Proxy == nil) {
+		shouldUpdateProxy = true
+	}
+	if state.Proxy != nil && plan.Proxy != nil {
```

Comment:
We can skip this if statement when `shouldUpdateProxy` is true (from line 1213). right?